### DISCUSSION TITLE
Player - song repeat

### DISCUSSION THREAD

**Discussion by ROBROSY** [FluidSynth/fluidsynth#1201] [Q&A]
In the **fluid_player_play** description:
"If the list of files added to the player has completed its requested number of loops, the playlist will be restarted from the beginning with a loop count of 1."

What I understand is that the file playback is repeated indefinitely. That doesn't happen with me. I can't even force replay, need to re-add the file via **fluid_player_add**.
Do you know what this description means? And is it possible to replay a "finished" file without re-**fluid_player_add** ?

**Comment by derselbst** [comment]
I would expect that the player starts the playlist from the beginning after another call to `fluid_player_play()` when finished. If you want infinite looping, see `fluid_player_set_loop()`.

**Comment by ROBROSY** [comment]
fluid_player_play() doesn't work. Only reloading via fluid_player_add causes playback.

I still don't know what that description means...

**Reply by derselbst** [reply]
> fluid_player_play() doesn't work. Only reloading via fluid_player_add causes playback.
> 
> I still don't know what that description means...

Well... it would really help a lot if you provided a complete minimal reproducible example in form of C source code. This is also true for the other issue you've opened. My time is currently heavily limited. I just need something that I can compile and look into when I find a few minutes of my spare time. I'll not spend this time inventing code that shows your problem, sry.

**Comment by ROBROSY** [comment]
#include <stdlib.h>
#include <string.h>
#include <fluidsynth.h>
 
/* An example midi file */
const char MIDIFILE[] = {
    0x4d, 0x54, 0x68, 0x64, 0x00, 0x00, 0x00, 0x06,
    0x00, 0x01, 0x00, 0x01, 0x01, 0xe0, 0x4d, 0x54,
    0x72, 0x6b, 0x00, 0x00, 0x00, 0x20, 0x00, 0x90,
    0x3c, 0x64, 0x87, 0x40, 0x80, 0x3c, 0x7f, 0x00,
    0x90, 0x43, 0x64, 0x87, 0x40, 0x80, 0x43, 0x7f,
    0x00, 0x90, 0x48, 0x64, 0x87, 0x40, 0x80, 0x48,
    0x7f, 0x83, 0x60, 0xff, 0x2f, 0x00
};
 
int main(int argc, char** argv)
{
    int i;
    void* buffer;
    size_t buffer_len;
    fluid_settings_t* settings;
    fluid_synth_t* synth;
    fluid_player_t* player;
    fluid_audio_driver_t* adriver;
    settings = new_fluid_settings();
    synth = new_fluid_synth(settings);
    player = new_fluid_player(synth);
    fluid_settings_setstr(settings, "audio.driver", "alsa");
    adriver = new_fluid_audio_driver(settings, synth);
    fluid_synth_sfload(synth, "/usr/share/sounds/sf2/default-GM.sf2", 1);
    /* queue up the in-memory midi file */
    fluid_player_add_mem(player, MIDIFILE, sizeof(MIDIFILE));
    /* play the midi file */
    fluid_player_play(player);
    /* wait for playback termination */
    fluid_player_join(player);
    fluid_player_seek(player,0);
    fluid_player_play(player);
    fluid_player_join(player);
    /* cleanup */
    delete_fluid_audio_driver(adriver);
    delete_fluid_player(player);
    delete_fluid_synth(synth);
    delete_fluid_settings(settings);
    return 0;
}

**Comment by derselbst** [comment]
I cannot reproduce your problem. The in-memory MIDI file of your example contains three notes. Calling play() and join() causes these three notes to be played. Calling play() and join() again causes the three notes to be played again. That's why in total I hear 6 notes. The seek() is not needed. Also, you're not "reloading" anything via fluid_player_add as you've described earlier.

**Comment by ROBROSY** [comment]
OK, it works better in the newer version.
But the instructions are incomprehensible...
